### PR TITLE
[GitHubGistStars] add GitHub Gist Stars

### LINKED ISSUE #8269: Badge request: stars for github gist
:clipboard: **Description**

Please add dynamic badge to show number of stars for github gist (not repo).

Example: [GitHub Gist stars](https://gist.github.com/maratori/47a4d00457a92aa426dbd48a18776322)

:link: **Data**

Looks like need to use GraphQL: https://docs.github.com/en/graphql/reference/objects#gist
```graphql
{
  user(login: "maratori") {
    gist(name: "47a4d00457a92aa426dbd48a18776322") {
      stargazerCount
    }
  }
}
```
As far as I understand, it requires authentication.

:microphone: **Motivation**

It'd be great to add this badge into my [github profile](https://github.com/maratori#-useful-gists).

### PR DESCRIPTION
**Pull request by notanmay**
closes #8269 
added a new service class which extends `GithubAuthV4Service` . It uses Github GraphQL API to query the stargazers count for the provided gist id. also added tester file for the same service class.

screen-shots of example badges generated:
https://gist.github.com/47a4d00457a92aa426dbd48a18776322
sss (1)

https://gist.github.com/invalid-gist-id
Screenshot from 2022-10-03 18-44-40 (1) (1)

### CODE REVIEW

File: services/github/github-gist-stars.service.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import { NotFound } from '../index.js'
+import { GithubAuthV4Service } from './github-auth-service.js'
+import { documentation as commonDocumentation } from './github-helpers.js'
+
+const schema = Joi.object({
+  data: Joi.object({
+    viewer: Joi.object({
+      gist: Joi.object({
+        stargazerCount: Joi.number(),
+        url: Joi.string(),
```
Comment (chris48s) [inline services/github/github-gist-stars.service.js:13]:
```suggestion
        stargazerCount: Joi.number().required(),
        url: Joi.string().required(),
```

File: services/github/github-gist-stars.service.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+const documentation = `${commonDocumentation}
+<p>This badge shows the number of stargazers for a gist. Gist id is accepted as input and 'gist not found' is returned if the gist is not found for the given gist id.
+</p>`
+
+export default class GithubGistStars extends GithubAuthV4Service {
+  static category = 'social'
+
+  static route = {
+    base: 'github/stars/gists',
+    pattern: ':gist',
```
Comment (chris48s) [inline services/github/github-gist-stars.service.js:28]: Can we make the param `:gistId` to match https://github.com/badges/shields/blob/8e1680d65d12b97356a587db929ecb80e874fd78/services/github/github-gist-last-commit.service.js#L13

File: services/github/github-gist-stars.service.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  ]
+
+  static defaultBadgeData = {
+    label: 'Stars',
+    color: 'blue',
+    namedLogo: 'github',
+  }
+
+  static render({ stargazerCount, url }) {
+    return { message: metric(stargazerCount), link: [url] }
```
Comment (chris48s) [inline services/github/github-gist-stars.service.js:51]: On the repo stars badge, we link the left side of the badge to the repo URL and the right side to `/stargazers`. Can we do the same thing for the gist version too please.
https://github.com/badges/shields/blob/8e1680d65d12b97356a587db929ecb80e874fd78/services/github/github-stars.service.js#L48-L51
Gists also have a similar stargazers URL e.g: https://gist.github.com/maratori/47a4d00457a92aa426dbd48a18776322/stargazers

**Review by chris48s — COMMENTED** [review]
Thanks for picking this up. It is in pretty good shape already. Just a few minor comments.

**Comment by notanmay** [comment]
hi @chris48s  , thanks for suggesting these changes. Please review the commit i just pushed. I have committed those changes and updated test cases as well and it passes the tests.

**Review by chris48s — APPROVED** [review]
Nice. Thanks for working on this.

**Comment by notanmay** [comment]
Thanks for merging my PR!🚀

**Pull request merged**